### PR TITLE
Use mono 4.0.1 instead of mono 4.1.0 according to aspnet/home, change vs...

### PR DESCRIPTION
**Pull request by yanchenw**
Use mono 4.0.1 instead of mono 4.1.0 as aspnet/home suggested:
https://github.com/aspnet/Home/blob/dev/GettingStartedDeb.md

Change vs-1.0.0-beta4 to based on official aspnet:1.0.0-beta4 to keep them consistent.

### CODE REVIEW

**Comment by msotclas** [comment]
Hi __@yanchenw__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla2.msopentech.com.
    </span>

TTYL, MSOTBOT;

**Comment by ahmetalpbalkan** [comment]
@yanchenw thanks for the PR! I will hold off until this PR until mono:4.1.0 is tagged on Docker Hub if there's no urgent need?

**Comment by yanchenw** [comment]
Should be OK. Any idea when mono 4.1.0 will be tagged?

**Comment by ahmetalpbalkan** [comment]
@yanchenw nope but once it appears at https://github.com/mono/mono/tags it'll probably take a day or so.

**Comment by akoeplinger** [comment]
Mono 4.1 (like all odd subversions) is the "in development" version and typically won't receive a tag.

**Comment by ahmetalpbalkan** [comment]
@akoeplinger Oh, I didn't know that convention. It is apparently case with mono Docker images as well: https://registry.hub.docker.com/u/library/mono/

I guess we could still use 4.1.0 release by doing `apt-get install mono` from Mono's repositories? In that case, this PR seems like the only solution.

File: vs-1.0.0-beta4/Dockerfile [hunk 1]
```diff
@@ -1,36 +1 @@
-FROM mono:3.12
```
Comment (ahmetalpbalkan) [inline vs-1.0.0-beta4/Dockerfile]: there's no need for this file anymore.

I will make this point to `1.0.0-beta4` folder on Docker Hub settings once merged and 1.0.0-beta4 and vs-1.0.0-beta4 will be the same image.

File: 1.0.0-beta4/Dockerfile [hunk 2]
```diff
@@ -1,20 +1,16 @@
...
-    && make install \
-    && cd .. \
-    && rm -rf mono
-
 ENV DNX_VERSION 1.0.0-beta4
 ENV DNX_USER_HOME /opt/dnx
 
+RUN apt-get -qq update && apt-get -qqy install unzip
+
+# Install Mono 4.0.1 or above to get bug fixes for ASP.NET.
```
Comment (ahmetalpbalkan) [inline 1.0.0-beta4/Dockerfile]: Is the `4.0.1` part accurate @yanchenw ? Shouldn't it be 4.1.0 (as it was above)?

**Comment by akoeplinger** [comment]
> I guess we could still use 4.1.0 release by doing  apt-get install mono  from Mono's repositories?

4.1 will never be in the repositories, it'll be declared 4.2 once it's "stable".

**Comment by ahmetalpbalkan** [comment]
@akoeplinger ok I was deceived by the `apt-get install mono-complete` @yanchenw does in the beta4 dockerfile... There's a `mono:4.0.1` why don't we just use that?

File: vs-1.0.0-beta4/Dockerfile (hunk 1 above)
Comment (yanchenw) [inline vs-1.0.0-beta4/Dockerfile]: Removed

**Comment by ahmetalpbalkan** [comment]
@yanchenw thanks for the change. Merging assuming you tested the product that uses the vs-* image.

**Comment by ahmetalpbalkan** [comment]
Waiving CLA since @yanchenw is a MSFT employee (he's registering himself to the CLA system now).

**Pull request merged**